src/map/ai/states/mobskill_state.cpp

```diff
@@ -109,6 +109,19 @@ bool CMobSkillState::Update(time_point tick)
             static_cast<CMobEntity*>(PTarget)->PEnmityContainer->UpdateEnmity(m_PEntity, 0, 0);
         }
         m_PEntity->PAI->EventHandler.triggerListener("WEAPONSKILL_STATE_EXIT", CLuaBaseEntity(m_PEntity), m_PSkill->getID());
+
+        if (m_PEntity->objtype == TYPE_PET && m_PEntity->PMaster && m_PEntity->PMaster->objtype == TYPE_PC && (m_PSkill->isBloodPactRage() || m_PSkill->isBloodPactWard()))
+        {
+            CCharEntity* PSummoner = (CCharEntity*)(m_PEntity->PMaster);
```

Comment:
Can you change this to

`CCharEntity* PSummoner = dynamic_cast<CCharEntity*>(m_PEntity->PMaster);`
It's much, much safer than a direct cast like that. It does voodoo to verify things on the stack to verify the class is infact that class, rather than "forcing" it into that class whether it is that or not.